src/hyperloglog.c

```diff
@@ -184,6 +188,14 @@ struct hllhdr {
 #define HLL_RAW 255 /* Only used internally, never exposed. */
 #define HLL_MAX_ENCODING 1
 
+/* Detect whether we can enable AVX2 optimizations. 
+ * The conditions are picked loosely and can be adjusted as needed. */
+#if (defined(__GNUC__) && __GNUC__ >= 7) || (defined(__clang__) && __clang_major__ >= 7)
+#if defined(__x86_64__)
+#define HLL_ENABLE_AVX2
+#endif
+#endif
+
```

Comment:
maybe you can refer to this PR https://github.com/redis/redis/pull/13609
move this check it config.h
also need to check `__attribute__((__target__("avx2")))`

Reply:
changed in e4242c993b6784d4fc14e70b0dfda317405a59ba